pkg/controller/replica.go

```diff
@@ -162,11 +165,33 @@ func (r *ReconcileReplica) triggerGarbageCollection(volumeName string) {
 func (r *ReconcileReplica) removeGarbageCollection(volumeName string) {
 	v, ok := r.cleanUpMap.LoadAndDelete(volumeName)
 	if ok {
-		cancelFunc := v.(context.CancelFunc)
-		cancelFunc()
+		entry := v.(cleanUpEntry)
+		entry.cancelFunc()
 	}
 }
 
+func (r *ReconcileReplica) retryManageReplicas(ctx context.Context, volumeName string) {
+	// retry replica management with exponential backoff until success
+	conditionFunc := func() (bool, error) {
+		// we should not run replica management while garbage collection is scheduled for the volume
+		if v, cleanUpScheduled := r.cleanUpMap.Load(volumeName); cleanUpScheduled {
+			entry := v.(cleanUpEntry)
+			select {
+			case <-entry.ctx.Done():
+				// if gargage collection cancelled, start replica management
+				break
+			case <-timeSinceEquals(entry.initTimestamp, r.timeUntilGarbageCollection):
+				// if garbage collection initiated, quit replica management
+				return true, nil
+			}
```

Comment:
Are you deliberately running a blocking `select` to wait for either cancelation or initiation?